spec/helpers/application_cocoon_helper_spec.rb

```diff
@@ -0,0 +1,66 @@
+require 'rails_helper'
+
+RSpec.describe ApplicationCocoonHelper, type: :helper do
+  class TestView < ActionView::Base
+    include ApplicationCocoonHelper
+  end
+
+  describe '#link_to_association' do
+    let!(:template) { TestView.new }
+    let(:course_group) { Course::Group.new }
+    let(:form_object) { double(object: course_group, object_name: course_group.class.name) }
+    let(:args) { ['Add group_user', form_object, :group_users] }
+    before do
+      allow(template).to receive(:render_association).and_return('form<tag>')
+    end
+
+    context 'when html options is not given' do
+      subject { template.link_to_association(*args) }
+      it 'returns the same result as link_to_add_association' do
+        expect(subject).to eq(template.link_to_add_association(*args))
+      end
+    end
+
+    context 'when html options is given' do
+      let(:html_options) do
+        {
+          anchor_node: :tbody,
+          insertion_method: :append,
+          traversal: :next
+        }
+      end
+
+      let(:old_options) do
+        {
+          'data-association-insertion-node' => 'tbody',
+          'data-association-insertion-method' => 'append',
+          'data-association-insertion-traversal' => 'next'
+        }
+      end
+      subject { template.link_to_association(*args, html_options) }
+
+      it 'returns the same result as link_to_add_association' do
```

Comment:
No, when cocoon changes its behaviour my tests will not pass anymore.

If I'm checking the resulting html, that is like testing cocoon. Because the html is generated by it.